efficientdet/keras/inference.py

```diff
@@ -0,0 +1,322 @@
+      ckpt_path: checkpoint path, such as /tmp/efficientdet-d0/.
+      batch_size: batch size for inference.
+      use_xla: Whether run with xla optimization.
+      min_score_thresh: minimal score threshold for filtering predictions.
+      max_boxes_to_draw: the maximum number of boxes per image.
+      model_params: model parameters for overriding the config.
+    """
+    super().__init__()
+    self.model_name = model_name
+    self.ckpt_path = ckpt_path
+    self.batch_size = batch_size
+
+    self.params = hparams_config.get_detection_config(model_name).as_dict()
+
+    if model_params:
+      self.params.update(model_params)
+    self.params.update(dict(is_training_bn=False))
+    self.label_map = self.params.get('label_map', None)
+
+    self.model = None
+
+    self.min_score_thresh = min_score_thresh
+    self.max_boxes_to_draw = max_boxes_to_draw
+    mixed_precision = self.params.get('mixed_precision', None)
+    precision = utils.get_precision(
+        self.params.get('strategy', None), mixed_precision)
+    policy = tf.keras.mixed_precision.experimental.Policy(precision)
+    tf.keras.mixed_precision.experimental.set_policy(policy)
+
+  def build(self, params_override=None):
+    """Build model and restore checkpoints."""
+    params = copy.deepcopy(self.params)
+    if params_override:
+      params.update(params_override)
+    config = hparams_config.get_efficientdet_config(self.model_name)
+    config.override(params)
+    self.model = EfficientDetModel(config=config)
+    image_size = utils.parse_image_size(params['image_size'])
+    self.model.build((self.batch_size, *image_size, 3))
+    util_keras.restore_ckpt(self.model, self.ckpt_path, params['moving_average_decay'])
+
+  def visualize(self, image, boxes, classes, scores, **kwargs):
+    """Visualize prediction on image."""
+    return visualize_image(image, boxes, classes.astype(int), scores,
+                           self.label_map, **kwargs)
+
+  def benchmark(self, image_arrays, bm_runs=10, trace_filename=None):
+    """Benchmark inference latency/throughput.
+
+    Args:
+      image_arrays: a list of images in numpy array format.
+      bm_runs: Number of benchmark runs.
+      trace_filename: If None, specify the filename for saving trace.
+    """
+    if not self.model:
+      self.build()
+
+    @tf.function
+    def test_func(image_arrays):
+      return self.model(image_arrays)
+
+    test_func(image_arrays)
+
+    start = time.perf_counter()
+    for _ in range(bm_runs):
+      test_func(image_arrays)
+    end = time.perf_counter()
+    inference_time = (end - start) / bm_runs
+
+    print('Per batch inference time: ', inference_time)
+    print('FPS: ', self.batch_size / inference_time)
+
+    if trace_filename:
+      options = tf.profiler.experimental.ProfilerOptions()
+      tf.profiler.experimental.start(trace_filename, options)
+      test_func(image_arrays)
+      tf.profiler.experimental.stop()
+
+  def serve_files(self, image_files: List[Text]):
```

Comment:
I find this is not useful at all, let's delete this "serve_files" to simplify the code.